src/components/numeral-date/numeral-date.test.tsx

```diff
@@ -0,0 +1,1995 @@
+      />
+    );
+
+    expect(ref.current).toBe(screen.getByRole("textbox", { name: "Year" }));
+  });
+
+  it("should be passed down to the `Year` input when it is as a callback", () => {
+    const ref = jest.fn();
+    render(
+      <NumeralDate
+        value={{ dd: "", mm: "", yyyy: "" }}
+        onChange={() => {}}
+        yearRef={ref}
+      />
+    );
+
+    expect(ref).toHaveBeenCalledWith(
+      screen.getByRole("textbox", { name: "Year" })
+    );
+  });
+
+  it("should set to empty after unmount", () => {
+    const ref = { current: null };
+    const { unmount } = render(
+      <NumeralDate
+        value={{ dd: "", mm: "", yyyy: "" }}
+        onChange={() => {}}
+        yearRef={ref}
+      />
+    );
+
+    unmount();
+
+    expect(ref.current).toBe(null);
+  });
+});
+
+test("should not call the onChange callback when the prop is set and the user types a value that exceeds the 'Day' input limit", async () => {
+  const onChange = jest.fn();
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+  render(
+    <NumeralDate value={{ dd: "12", mm: "", yyyy: "" }} onChange={onChange} />
+  );
+  const dayInput = screen.getByRole("textbox", { name: "Day" });
+  await user.click(dayInput);
+  await user.keyboard("{End}");
+  await user.type(dayInput, "3");
+
+  expect(onChange).not.toHaveBeenCalled();
+});
+
+test("should not call the onChange callback when the prop is set and the user types a value that exceeds the 'Month' input limit", async () => {
+  const onChange = jest.fn();
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+  render(
+    <NumeralDate value={{ dd: "", mm: "12", yyyy: "" }} onChange={onChange} />
+  );
+  const monthInput = screen.getByRole("textbox", { name: "Month" });
+  await user.click(monthInput);
+  await user.keyboard("{End}");
+  await user.type(monthInput, "3");
+
+  expect(onChange).not.toHaveBeenCalled();
+});
+
+test("should not call the onChange callback when the prop is set and the user types a value that exceeds the 'Year' input limit", async () => {
+  const onChange = jest.fn();
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+  render(
+    <NumeralDate value={{ dd: "", mm: "", yyyy: "2011" }} onChange={onChange} />
+  );
+  const yearInput = screen.getByRole("textbox", { name: "Year" });
+  await user.click(yearInput);
+  await user.keyboard("{End}");
+  await user.type(yearInput, "3");
+
+  expect(onChange).not.toHaveBeenCalled();
+});
+
+test("should set the passed `data-` props as atributes on the root element", () => {
+  render(
+    <NumeralDate
+      value={{ dd: "", mm: "", yyyy: "" }}
+      onChange={() => {}}
+      data-component="numeral-date-root"
+      data-element="numeral-date-element"
+      data-role="numeral-date-role"
+    />
+  );
+  const rootElement = screen.getByTestId("numeral-date-role");
+
+  expect(rootElement).toHaveAttribute("data-component", "numeral-date-root");
```

Comment:
**comment** (non-blocking): rather surprised that we allow `data-component` to  be overr-idden, and not even as a private prop. But apparently we do! 😱 (In which case the test is fair enough.)

Reply:
yep agree, I've chosen to avoid making such a trivial breaking change for now as I think we should batch ti with other ones etc, I'll raise something in jira for it once I've reviewed if we have other components doing the same (we likely do 😞 )